lib/ci-stack.ts

```diff
@@ -33,19 +33,11 @@ export class CIStack extends Stack {
       },
     });
 
-    const useSslParameter = new CfnParameter(this, 'useSsl', {
-      description: 'If the jenkins instance should be access via SSL',
-      allowedValues: ['true', 'false'],
+    const useSslParameter = this.node.tryGetContext('useSsl');
+    const useSsl = useSslParameter === 'true';
 
-    });
-
-    const runWithOidcParameter = new CfnParameter(this, 'runWithOidc', {
-      description: 'If the jenkins instance should use OIDC + federate',
-      allowedValues: ['true', 'false'],
-    });
-
-    const useSsl = useSslParameter.valueAsString === 'true';
-    const runWithOidc = runWithOidcParameter.valueAsString === 'true';
+    const runWithOidcParameter = this.node.tryGetContext('runWithOidc');
+    const runWithOidc = runWithOidcParameter === 'true';
```

Comment:
Glad it worked!
You need to add checks if the values entered by user is either true or false. CfnParameters does that by using `allowedValues`, for context variables we need to add checks AFAIK

Reply:
Is that provision allowed with tryGetContext()? CfnParameters could not be used since we needed the actual values at runtime.